### ISSUE TITLE
tree-view improve styling and compatible with dialogs

### ISSUE THREAD

**Issue by FranGomezVenegas** [FranGomezVenegas/FrontE-Issues#357]
The view having the example is this one: 
![image](https://github.com/FranGomezVenegas/BackE-Issues/assets/26067165/8e7e723b-96db-4efd-b6c9-6885f4255fe7)

There are two components involved on this: tree-view and tree-node, both upon the directory called Tree:
![image](https://github.com/FranGomezVenegas/BackE-Issues/assets/26067165/80747470-5412-4b43-8da4-bc6a77d445d4)

Context on how it works:
The objects having child-records are marked with this extra symbol (+) that toggles the ability of expanding/collapsing the child level.

- [ ] The first thing to do is on apply the styling we already have, the only one component 'plugged' to the themes is readOnlyTable (line 2631 within DataViews.js, the theme is applied between line 2647 and 2651). Please review the mechanism by which the theme is applied and let us know how it looks like in your opinion, we need to have the best option for themes and then plug all components/elements to it. The Panetone about the styling for TRAZiT is on this link:  https://1drv.ms/b/s!Ah1ARh7IPhH8gcdMDMvoHUdJ0s3pig?e=LM2yId

**We should start ensuring the styling is applied everywhere and the theme artifact should help us on this goal**
We assume this is not one '5 minutes thing' therefore please lets monitoring this stuff all together as a team, please any doubt, anything you have no a clear perception you are doing it in the right way please let me know.

- [x] Then the second thing: The selected value. The user can select one entry at the parent level or even child level, only one. Click one for the first time means select it. If the user clicks the same one again then it means unselect it. If the user clicks another one then this new one becomes the selected one. This behavior is already done and the alert will help you to follow it. The key point is not the behaviour, the key point is on how to highlight the selected one to let the user be aware about which is the selected and even if nothing is highlighted then to assume that nothing is currently selected.

- [ ] The third thing: Styling. the text look&feel should be improved even, maybe, the symbol (+/-) to open/close child level.

- [ ] The last one: Add this feature to the genericDialog. Please ping me before starting this point to guide you on how to do it.

- [ ]
